src/code/z_map_disp.c

```diff
@@ -1,109 +1,1877 @@
+    if (((play->csCtx.state != CS_STATE_IDLE) && !MapDisp_IsLocationRomaniRanchCutscene(play)) ||
+        (sMapDisp.unk20 & 2) || Map_IsInBossScene(play)) {
+        return true;
+    }
+    return false;
+}
+
+s32 MapDisp_IsMinimapToggleBlocked(PlayState* play) {
+    if ((MapDisp_IsLocationMinimapBlocked(play) == true) || !MapDisp_CanDisplayMinimap(play)) {
+        return true;
+    }
+    return false;
+}
+
+s32 MapDisp_AreRoomsSameStorey(s32 curRoom, s32 prevRoom) {
+    MapDataRoom* mapDataRoom;
+    s16* roomStoreyList;
+
+    if ((curRoom == -1) || (prevRoom == -1)) {
+        return false;
+    }
+    mapDataRoom = &sMapDisp.mapDataScene->rooms[curRoom];
+    roomStoreyList = sMapDisp.roomStoreyList;
+    if ((roomStoreyList[curRoom] <= roomStoreyList[prevRoom]) &&
+        (roomStoreyList[prevRoom] <= (roomStoreyList[curRoom] + MAP_DATA_ROOM_GET_EXTRA_STOREYS(mapDataRoom)))) {
+        return true;
+    }
+    mapDataRoom = &sMapDisp.mapDataScene->rooms[prevRoom];
+    if ((roomStoreyList[prevRoom] <= roomStoreyList[curRoom]) &&
+        (roomStoreyList[curRoom] <= (roomStoreyList[prevRoom] + MAP_DATA_ROOM_GET_EXTRA_STOREYS(mapDataRoom)))) {
+        return true;
+    }
+    return false;
+}
+
+void MapDisp_DrawMinimap(PlayState* play, s32 playerInitX, s32 playerInitZ, s32 playerInitDir) {
+    PauseContext* pauseCtx = &play->pauseCtx;
+
+    if ((sMapDisp.mapDataScene != NULL) && ((s32)pauseCtx->state <= PAUSE_STATE_OPENING_2) && !R_MINIMAP_DISABLED &&
+        (play->interfaceCtx.minimapAlpha != 0)) {
+        if (!MapDisp_IsLocationMinimapBlocked(play) && (sSceneNumRooms != 0)) {
+            if (MapDisp_CanDisplayMinimap(play)) {
+                MapDisp_DrawMinimapRoom(play, sMapDisp.minimapCurTex, sMapDisp.minimapCurX, sMapDisp.minimapCurY,
+                                        sMapDisp.curRoom, 1.0f - (sMapDisp.swapAnimTimer * 0.05f));
+                if ((sMapDisp.curRoom != sMapDisp.prevRoom) &&
+                    MapDisp_AreRoomsSameStorey(sMapDisp.curRoom, sMapDisp.prevRoom)) {
+                    MapDisp_DrawMinimapRoom(play, sMapDisp.minimapPrevTex, sMapDisp.minimapCurX + sMapDisp.minimapPrevX,
+                                            sMapDisp.minimapCurY + sMapDisp.minimapPrevY, sMapDisp.prevRoom,
+                                            sMapDisp.swapAnimTimer * 0.05f);
+                }
+                MapDisp_Minimap_DrawDoorActors(play);
+            }
+            if ((!MapExp_CurRoomHasMapI(play) || CHECK_DUNGEON_ITEM(DUNGEON_COMPASS, gSaveContext.mapIndex)) &&
+                (MapExp_CurRoomHasMapI(play) || Inventory_IsMapVisible(play->sceneId))) {
+                if (play->interfaceCtx.minigameState == MINIGAME_STATE_NONE) {
+                    MapDisp_Minimap_DrawRedCompassIcon(play, playerInitX, playerInitZ, playerInitDir);
+                }
+                MapDisp_Minimap_DrawActors(play);
+            }
+        }
+    }
+}
+
+void MapDisp_ResetIMap(void) {
```

Comment:
Should this be `MapDisp_ResetMapI` to match the other functions and struct members.